ambry-server/src/integration-test/java/com/github/ambry/server/AccountStatsMySqlStoreIntegrationTest.java

```diff
@@ -0,0 +1,157 @@
+      boolean withLocalbackup) throws Exception {
+    Path localBackupFilePath = withLocalbackup ? createLocalBackup(10, 10, 1) : createTemporaryFile();
+    Properties configProps = Utils.loadPropsFromResource("mysql.properties");
+    configProps.setProperty(ClusterMapConfig.CLUSTERMAP_CLUSTER_NAME, clustername);
+    configProps.setProperty(ClusterMapConfig.CLUSTERMAP_HOST_NAME, hostname);
+    configProps.setProperty(ClusterMapConfig.CLUSTERMAP_DATACENTER_NAME, "dc1");
+    configProps.setProperty(ClusterMapConfig.CLUSTERMAP_PORT, String.valueOf(port));
+    configProps.setProperty(AccountStatsMySqlConfig.DOMAIN_NAMES_TO_REMOVE, ".github.com");
+    configProps.setProperty(StatsManagerConfig.STATS_OUTPUT_FILE_PATH, localBackupFilePath.toString());
+    VerifiableProperties verifiableProperties = new VerifiableProperties(configProps);
+    return new AccountStatsMySqlStoreFactory(verifiableProperties, new ClusterMapConfig(verifiableProperties),
+        new StatsManagerConfig(verifiableProperties), new MetricRegistry()).getAccountStatsMySqlStore();
+  }
+
+  private static Path createLocalBackup(int numPartitions, int numAccounts, int numContainers) throws IOException {
+    Path localBackupFilePath = createTemporaryFile();
+    ObjectMapper objectMapper = new ObjectMapper();
+    objectMapper.writeValue(localBackupFilePath.toFile(),
+        generateStatsWrapper(numPartitions, numAccounts, numContainers));
+    return localBackupFilePath;
+  }
+
+  private static Path createTemporaryFile() throws IOException {
+    Path tempDir = Files.createTempDirectory("AccountStatsMySqlStoreTest");
+    return tempDir.resolve("localbackup");
+  }
+
+  private static StatsWrapper generateStatsWrapper(int numPartitions, int numAccounts, int numContainers) {
+    Random random = new Random();
+    List<StatsSnapshot> storeSnapshots = new ArrayList<>();
+    for (int i = 0; i < numPartitions; i++) {
+      storeSnapshots.add(
+          TestUtils.generateStoreStats(numAccounts, numContainers, random, StatsReportType.ACCOUNT_REPORT));
+    }
+    return TestUtils.generateNodeStats(storeSnapshots, 1000, StatsReportType.ACCOUNT_REPORT);
+  }
+
+  private void cleanup(MySqlDataAccessor dataAccessor) throws SQLException {
+    Connection dbConnection = dataAccessor.getDatabaseConnection(true);
+    Statement statement = dbConnection.createStatement();
+    statement.executeUpdate("DELETE FROM " + AccountReportsDao.ACCOUNT_REPORTS_TABLE);
+  }
+
+  private void assertTableSize(MySqlDataAccessor dataAccessor, int expectedNumRows) throws SQLException {
+    Connection dbConnection = dataAccessor.getDatabaseConnection(true);
+    Statement statement = dbConnection.createStatement();
+    ResultSet resultSet = statement.executeQuery("SELECT * FROM " + AccountReportsDao.ACCOUNT_REPORTS_TABLE);
+    int numRows = 0;
+    if (resultSet != null) {
+      while (resultSet.next()) {
+        numRows++;
+      }
+    }
+    resultSet.close();
```

Comment:
This may cause NPE.

Reply:
Feel free to move closeQuietly() from MySqlDataAccessor to MySqlUtils and call it from here.

Reply:
updated.